File: neetcode/49. Group Anagrams/49.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map<String, List<String>> map = new HashMap<>();

        for (String s : strs) {
            char[] tmp = s.toCharArray();

            Arrays.sort(tmp);

            String sortedString = new String(tmp);

            if (!map.containsKey(sortedString)) {
                map.put(sortedString, new ArrayList<>());
            }
            map.get(sortedString).add(s);
        }

        return new ArrayList<>(map.values());
    }
}

File: neetcode/49. Group Anagrams/49.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map : MutableMap<String, MutableList<String>> = mutableMapOf()

        for (s in strs) {
            val tmp = s.toCharArray().sorted().joinToString("")

            map.getOrPut(tmp){ mutableListOf() }.add(s)
        }

        return map.values.toList()
    }
}
